src/main/java/org/generationcp/ibpworkbench/germplasm/pedigree/CreatePedigreeGraph.java

```diff
@@ -138,36 +139,34 @@ private void addNode(final GermplasmPedigreeTreeNode node, final int level) {
 			final String leafNodeGIDRoot = this.createNodeTextWithFormatting(node);
 			this.graphVizUtility.addln(leafNodeGIDRoot + ";");
 		}
-
 		for (final GermplasmPedigreeTreeNode parent : node.getLinkedNodes()) {
-			final boolean isFemaleParent = parent.getGermplasm().getGid().equals(node.getGermplasm().getGpid1())
-				&& node.getGermplasm().getGnpgs() >= 2;
-			final boolean isDerivative = node.getGermplasm().getGnpgs().equals(-1);
 			final String leafNodeGID = this.createNodeTextWithFormatting(parent);
 			final String parentNodeGID = this.createNodeTextWithFormatting(node);
 
 			if (level == 1) {
 				final String leafNodeGIDRoot = this.createNodeTextWithFormatting(node);
-				this.graphVizUtility.addln(leafNodeGID + "->" + leafNodeGIDRoot + this.getArrowStyleString(isFemaleParent, isDerivative));
+				this.graphVizUtility.addln(leafNodeGID + "->" + leafNodeGIDRoot + this.getArrowStyleString(node.getGermplasm(), parent.getGermplasm()));
 			} else {
-				this.graphVizUtility.addln(leafNodeGID + "->" + parentNodeGID + this.getArrowStyleString(isFemaleParent, isDerivative));
+				this.graphVizUtility.addln(leafNodeGID + "->" + parentNodeGID + this.getArrowStyleString(node.getGermplasm(), parent.getGermplasm()));
 			}
 
 			this.addNode(parent, level + 1);
 		}
 	}
 
-	String getArrowStyleString(final boolean isFemaleParent, final boolean isDerivative) {
-		if(isFemaleParent) {
-			return " [color=\"RED\", arrowhead=\"odottee\"];";
-		} else if(!isDerivative) {
-			return " [color=\"BLUE\", arrowhead=\"veeodot\"];";
-		} else {
-			return  ";";
+	public String getArrowStyleString(Germplasm nodeGermplasm, Germplasm parentGermplasm) {
+		if (nodeGermplasm.getGnpgs() != null && nodeGermplasm.getGnpgs() >= 2 ) {
```

Comment:
gnpgs is non-nullable so no need for null checking

Reply:
@darla-leafnode , 

When node is Unknown gnpgs value is null, added checking to prevent null pointer exception when checking value.